Source/ServerTrustEvaluation.swift

```diff
@@ -326,6 +327,137 @@ public final class PublicKeysTrustEvaluator: ServerTrustEvaluating {
     }
 }
 
+/// Uses the pinned public keys to validate the server trust. The server trust is considered valid if one of the pinned
+/// public keys match one of the server certificate public keys. By validating both the certificate chain and host,
+/// public key pinning provides a very secure form of server trust validation mitigating most, if not all, MITM attacks.
+/// Applications are encouraged to always validate the host and require a valid certificate chain in production
+/// environments.
+public final class SubjectPublicKeysInfoTrustEvaluator: ServerTrustEvaluating {
+    private let keys: [String]
+    private let performDefaultValidation: Bool
+    private let validateHost: Bool
+
+    /// Creates a `SubjectPublicKeysInfoTrustEvaluator`.
+    ///
+    /// - Note: Default and host validation will fail when using this evaluator with self-signed certificates. Use
+    ///         `PinnedCertificatesTrustEvaluator` if you need to use self-signed certificates.
+    ///
+    /// - Parameters:
+    ///   - keys:                     The Base64 encoded SPKI hashes to use to validate public keys.
+    ///   - performDefaultValidation: Determines whether default validation should be performed in addition to
+    ///                               evaluating the pinned certificates. Defaults to `true`.
+    ///   - validateHost:             Determines whether or not the evaluator should validate the host, in addition to
+    ///                               performing the default evaluation, even if `performDefaultValidation` is `false`.
+    ///                               Defaults to `true`.
+    public init(keys: [String],
+                performDefaultValidation: Bool = true,
+                validateHost: Bool = true) {
+        self.keys = keys
+        self.performDefaultValidation = performDefaultValidation
+        self.validateHost = validateHost
+    }
+
+    private enum Asn1Header {
```

Comment:
Consider renaming this to `PublicKeyType`, renaming `bytes` to `headerBytes` and adding a `keySize: Int` var.

You can then have `PublicKeyType` be `RawRepresentable` by having the `init(rawValue: )` implementation check against the known `kSecAttrKeyType` values.

This would allow you to get rid of `getAsn1HeaderBytes` and have this instead in `hashSubjectPublicKeyInfoFromCertificate`:

```swift
       // ...
        guard let publicKeyAttributes = SecKeyCopyAttributes(publicKey) as? [CFString:Any],
            let rawPublicKeyType = publicKeyAttributes[kSecAttrKeyType] as? String,
            let publicKeyType = PublicKeyType(rawValue: rawPublicKeyType),
            let publicKeysize = publicKeyAttributes[kSecAttrKeySizeInBits] as? NSNumber,
            publicKeyType.keySize == publicKeysize
            else { return nil }
            // ...
```